katello/katello-backup

```diff
@@ -54,49 +95,85 @@ def run_cmd(command, exit_codes=[0])
   end
 end
 
-def backup_db_online
-  puts "Backing up postgres online schema... "
-  run_cmd("runuser - postgres -c 'pg_dumpall -g > #{@dir}/pg_globals.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc foreman > #{@dir}/foreman.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc candlepin > #{@dir}/candlepin.dump'")
-  puts "Done."
+def snapshot_backup
+  snapsize = @options[:snapshot_size] || "2G"
+  mountdir = "#{@options[:snapshot_mount]}/rootsnapshot" || "/mnt/snap/rootsnapshot"
+  FileUtils.mkdir_p mountdir
+  run_cmd("katello-service stop #{EXCLUDED}") unless @options[:online]
+  run_cmd("lvcreate -s -nrootsnapshot -L#{snapsize} /dev/#{@options[:snapshot]}/root")
```

Comment:
* this will fail if there is already a snapshot called `rootsnapshot` (which the user could have created them-self for some reason)
* this implies that all data is on the same LV with the name `root`. a prod deployment usually will have `/var`, `/var/lib/pgsql/`, `/var/lib/mongodb` and `/var/lib/pulp` on separate LVs, with varying names.